Perl/Perl.sublime-syntax

```diff
@@ -1490,31 +1506,38 @@ contexts:
         - match: \)
           scope: punctuation.section.parens.end.perl
           pop: true
-        - include: literal-parens-nested
-        - include: literal-common
+        - include: literal-parens-common
 
   literal-parens-nested:
     - match: \(
       push:
         - match: \)
           pop: true
-        - include: literal-parens-nested
-        - include: literal-common
+        - include: literal-parens-common
+
+  literal-parens-common:
+    - match: \\[\\\)]
+      scope: constant.character.escape.perl
+    - include: literal-parens-nested
+    - include: literal-common
 
   literal-generic-pop:
     - match: ({{regexp_delim}})
       scope: punctuation.section.generic.begin.perl
       set:
         - meta_scope: meta.generic.perl
         - meta_content_scope: meta.string.perl string.unquoted.perl
-        - match: \1
+        # Need a lookbehind here to avoid popping on escaped end-of-string token
+        # as a normal `\\\1` rule fails to run. As backrefs use oniguruma anyway
+        # it shouldn't hurt performance much.
```

Comment:
No, because 

a) `literal-common` doesn't contain any escapes.
b) in single quoted strings only the `\` and the terminating character may be escaped. In this special case any arbritary character (`\1` backref) is used to terminate the string and thus is allowed to be escaped. This is the only working solution I have found so far which works in conjunction with backrefs.

Reply:
My hope is actually to use some of the recent regex compilation optimizations for sregex in a future release to be able to use sregex for parameterized pops. Not that I think a single rule here will make a huge difference, just something to think about.